src/main/java/duke/DukeController.java

```diff
@@ -0,0 +1,121 @@
+package duke;
+
+import exception.EmptyTodoException;
+import exception.UnknownCommandException;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.ToDo;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class DukeController {
+
+    public static final int EVENT_STRING_LENGTH = 5;
+    public static final int DEADLINE_STRING_LENGTH = 8;
+    public static final int TODO_STRING_LENGTH = 4;
+    public static final int DELETE_STRING_LENGTH = 6;
+    public static ArrayList<Task> tasks = new ArrayList<>();
+    public static int taskCount = 0;
+
+    public static void run(Scanner scanner) {
+        tasks = DukeReader.getTaskListFromFile();
+        taskCount = tasks.size();
+        while (true) {
+            String input = scanner.nextLine();
+            if (input.equals("bye")) {
+                return;
+            } else if (input.equals("list")) {
+                UI.printTaskList(tasks, taskCount);
+            } else if (input.contains("done")) {
+                markTaskDone(input);
+            } else if (input.contains("delete")) {
+                deleteTask(input);
```

Comment:
You could consider declaring these magic strings (bye, list, done and delete) as constants the way you have done for string lengths. Similar changes can be made in other lines as well.